### PR TITLE
Errorhandler middleware

### PR DESCRIPTION
**Pull request by tulma95**

### CODE REVIEW

File: lib/utils.ts [hunk 1]
```diff
@@ -1,28 +1,43 @@
...
 export const parseLocale = (locale: string | string[]): string =>
   locale ? locale.toString() : 'en'
 
+export const validate = <A, I, O>(
+  runtimeType: t.Type<A, O, I>,
+  inputToValidate: I,
+): A => {
+  const errorOrResult = runtimeType.decode(inputToValidate)
+  if (isLeft(errorOrResult)) {
+    throw new TypeError(PathReporter.report(errorOrResult).toString())
```
Comment (karki) [inline lib/utils.ts:35]: Is this a meaningful error message for the end user?

Comment (tulma95) [reply lib/utils.ts:35]: It's not meaningful for end user. We can create new error type for io-ts errors and handle them separately in error handler middleware to send better error message for end user.

File: lib/errors.ts [hunk 2]
```diff
@@ -9,3 +9,9 @@ export class TicketNotFoundError extends Error {
     super(message)
   }
 }
+
+export class IotsError extends Error {
```
Comment (karki) [inline lib/errors.ts:13]: Maybe IoTsError

File: lib/middleware.ts [hunk 3]
```diff
@@ -43,6 +44,10 @@ export const withErrorHandler = (
       return res.status(400).json({ message: error.message })
     if (error instanceof TicketNotFoundError)
       return res.status(404).json({ message: error.message })
+    if (error instanceof IotsError)
+      return res
+        .status(400)
+        .json({ message: 'Io-ts runtime type validation error' })
```
Comment (karki) [inline lib/middleware.ts:50]: I'd abstract this even more. Maybe just `Invalid parameters` or something.

**Pull request merged**